### PR TITLE
tests(compute-static): add basic unit tests

### LINKED ISSUE #401: Write unit tests for `static.crates.io` function
## Context

The crates on crates.io are stored in S3 and served to users through our Content Delivery Networks, most importantly Fastly. On Fastly, we are using their [Compute](https://www.fastly.com/documentation/guides/compute/) platform to run every request through a [custom Rust function](https://github.com/rust-lang/simpleinfra/tree/master/terragrunt/modules/crates-io/compute-static).

This function mainly has two responsibilities:

1. It handles the request, which means applying some rewrite rules and fetching the crate from S3.
2. It logs the request and sends it to the configured log destinations.

Since the function runs in WebAssembly, testing the function has been historically difficult. But Fastly provides more documentation now that might make it feasible to write unit tests for (some parts of) the function.

## Tasks

- [ ] Write (a few) unit tests for the function
- [ ] Document how to run them in local development
- [ ] Run the tests automatically as part of this repository's CI

## Resources

- [How to write Rust unit tests for your Compute application](https://www.fastly.com/blog/how-to-write-rust-unit-tests-for-your-compute-edge-application)
- [Rust on the Compute platform > Unit testing](https://www.fastly.com/documentation/guides/compute/rust/#unit-testing)
- https://github.com/fastly/Viceroy#usage-as-a-test-runner

### PR DESCRIPTION
**Pull request by Carbonhell**
Closes #401 .

This PR adds a few basic unit tests with [Viceroy](https://github.com/fastly/Viceroy), a local runner for Fastly compute functions. There are a few functions I couldn't test:

- `set_ttl`: there's no way to retrieve the TTL set on a Request struct, and the function itself is trivial;
- ~~`send_request_to_s3`: this function sends a HTTP request to the backend, and Viceroy does not allow intercepting them (see [related issue](https://github.com/fastly/Viceroy/issues/442)). This also prevents implementing a test for `handle_request`.~~ out of scope for this PR (needs to be tested with an integration test due to wasm)

While working on this, I noticed the build target was still set to `wasm32-wasi` (see [here](https://github.com/rust-lang/compiler-team/issues/607)), which was breaking builds due to the target not being found - I've renamed it accordingly.
~~Also, I couldn't build the function either with the latest version of Rust (1.91) due to [an ongoing incident with Fastly](https://www.fastlystatus.com/incident/378001). I've pinned the version to 1.90 to allow builds, again.~~ The incident was resolved, so I reverted the toolchain to stable.

The tests have been integrated in the CI as part of the "Rust projects" job, although this step takes slightly less than 6 minutes due to the installation of Viceroy, mostly. If that's too much time, I can look into [caching](https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci), or we can split it to only download those tools and execute the tests when files change in the compute-static directory with [dorny/paths-filter](https://github.com/dorny/paths-filter).

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -23,6 +23,13 @@ jobs:
       - name: Run clippy
         run: cargo clippy --all -- -Dwarnings
 
+      - name: Run compute-static tests
+        working-directory: terragrunt/modules/crates-io/compute-static
+        run: |
+          curl -LsSf https://get.nexte.st/latest/linux | tar zxf - -C ${CARGO_HOME:-~/.cargo}/bin
```
Comment (syphar) [inline .github/workflows/main.yml]: I would recommend using `taiki-e/install-action` for this: 
```
- uses: taiki-e/install-action@v2
  with:
    tool: cargo-nextest,viceroy
```

Or directly `cargo binstall`

File: terragrunt/modules/crates-io/compute-static/scripts/run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Start the Python HTTP server in the background, which will act as primary host for requests sent from Rust tests
+python3 scripts/test_http_server.py &
```
Comment (syphar) [inline terragrunt/modules/crates-io/compute-static/scripts/run_tests.sh]: the "real" way to write integration tests is actually another one, using [`viceroy-lib`](https://docs.rs/viceroy-lib/latest/viceroy_lib/), directly, running the WASM binary. It's a little more complex, needs a non-wasm subcrate for the binary, bun I once made it work for a project then you could just  use `mockito` to generate a test case specific mock-server. And it's a "real" integration test. 

My proposal for this PR would be: 
Keep it simple and just for unit-tests. It's already super helpful to be able to write normal unit-tests, without a real server. This small python server has only limited value for testing the many edge cases you want to test.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Carbonhell) [reply .github/workflows/main.yml]: Right, didn't know about this action, thank you!

File: terragrunt/modules/crates-io/compute-static/scripts/run_tests.sh (hunk 2 above)
Comment (Carbonhell) [reply terragrunt/modules/crates-io/compute-static/scripts/run_tests.sh]: I agree. I consider this implementation with the Python HTTP server hacky. I initially implemented it as I would've preferred to add coverage for the `send_request_to_s3` function, as it's the common case handled by this compute function.
I didn't think of using viceroy-lib directly, that seems definitely cleaner. I just saw it's also mentioned in [Viceroy's README](https://github.com/fastly/Viceroy?tab=readme-ov-file#usage-as-a-library) as the suggested way to run integration tests. Considering the need for a subcrate for this kind of test, I also think it might be worth considering whether this is actually needed, and if so having a separate PR - I'll let the maintainers evaluate whether they'd appreciate this or if it's overkill for now.
